src/modelinstance.cpp

```diff
@@ -879,6 +890,17 @@ const Status ModelInstance::validate(const tensorflow::serving::PredictRequest*
         if (requestInput.dtype() == tensorflow::DataType::DT_STRING) {
             // binary inputs will be validated during conversion to blob
             SPDLOG_DEBUG("Received request contains binary inputs");
+            if (checkBinaryInputBatchSizeMismatch(*networkInput, requestInput)) {
+                if (batchingMode == AUTO) {
+                    return StatusCode::BATCHSIZE_CHANGE_REQUIRED;
+                } else {
```

Comment:
Please add another unit test for else condition:
requestInput.dtype() == tensorflow::DataType::DT_STRING
batchingMode != AUTO